addons/bus/models/ir_websocket.py

```diff
@@ -82,54 +40,40 @@ def _prepare_subscribe_data(self, channels, last):
             A dict containing the following keys:
             - channels (set of str): The list of channels to subscribe to.
             - last (int): The last known notification id.
-            - missed_presences (odoo.models.Recordset): The missed presences.
 
         :raise ValueError: If the list of channels is not a list of strings.
         """
         if not all(isinstance(c, str) for c in channels):
             raise ValueError("bus.Bus only string channels are allowed.")
         # sudo - bus.bus: reading non-sensitive last bus id.
         last = 0 if last > self.env["bus.bus"].sudo()._bus_last_id() else last
-        str_presence_channels = {
-            c for c in channels if isinstance(c, str) and c.startswith("odoo-presence-")
-        }
-        presence_channels = self._build_presence_channel_list(
-            [tuple(c.replace("odoo-presence-", "").split("_")) for c in str_presence_channels]
-        )
-        # There is a gap between a subscription client side (which is debounced)
-        # and the actual subcription thus presences can be missed. Send a
-        # notification to avoid missing presences during a subscription.
-        domain = expression.AND(
-            [
-                [("last_poll", ">", datetime.now() - timedelta(seconds=2))],
-                expression.OR(self._get_missed_presences_identity_domains(presence_channels)),
-            ]
-        )
-        # sudo: bus.presence: can access presences linked to presence channels.
-        missed_presences = self.env["bus.presence"].sudo().search(domain)
-        all_channels = OrderedSet(presence_channels)
-        all_channels.update(
-            self._build_bus_channel_list([c for c in channels if c not in str_presence_channels])
-        )
-        return {"channels": all_channels, "last": last, "missed_presences": missed_presences}
+        return {"channels": channels, "last": last}
```

Comment:
That's incorrect (and the docstring as well): previously, channels were computed by `_build_bus_channel_list`. The result was a set of records/tuples. Here, you only pass the `channels` parameter. However, most of the channels are added by `_build_bus_channel_list` overrides (according to the current user/guest).

Also, your override in the discuss module does not take into account super channels.

Reply:
> Also, your override in the discuss module does not take into account super channels.

The override of `_prepare_subscribe_data` doesn't need to take into account super channels since it's building the  channel list with `_build_bus_channel_list` and that takes into account the super, right?

Reply:
It looks like a very bad design to just ignore what `super` did. What if another module, called before this override adds channels coming from somewhere else than `_build_bus_channel_list`? To me, the correct way to do so is something like this:

```py
data["channels"] |= my_module_channels
```